src/components/NcButton/NcButton.vue

```diff
@@ -485,15 +635,16 @@ export default {
 	 * @return {object|undefined} The created VNode
 	 */
 	render(h) {
-		const text = this.$slots.default?.[0]?.text?.trim?.()
-
+		/** @slot **deprecated** Use the `text` property instead */
+		const text = this.text || this.$slots.default?.[0]?.text?.trim?.()
```

Comment:
Can we directly remove the slot in `master` which will be `8.x` and introduce the deprecation for `stable7`?

Reply:
This will be huge in terms of breaking projects, not sure if we want to do this already for version 8 or wait for the next

Reply:
We have huge breaking changes in master already (e.g. the renaming from `title` to `name`) and the entire removal of some components. So, from my side, I wouldn't mind another breaking change.
The migration effort needs to be done anyway, the question is just, when does it suite you 😉

Reply:
IMO, we should either allow using slot for button content. It would also solve custom text formatting problem by allowing to pass formatted text as a markup.
```vue
<NcButton>
  <span class="foo">My text</span> <span class="down">🔻</span>
</NcButton>
```

Or remove the content slot at all and make this breaking change with text.

During `beta` we can add check to `mounted` that `console.warn` if slot is passed to help with the migration.

Reply:
After seeing all of your validator fixes @ShGKme  I think we should go with this one and only allow text content through props.

Reply:
Pushed a commit that warns when the default slot is used as suggested by @ShGKme 
Also added a test to remind us to remove that behavior when releasing (as soon as the version is set to stable it will fail if the default slot still works :wink: )

Reply:
> After seeing all of your validator fixes @ShGKme I think we should go with this one and only allow text content through props.

For the record, the only place where `<button>` had invalid non-phrasal content was User Menu avatar-button-trigger, and plain html `<button>` was used there. There was no HTML validation issue with `NcButton` content.

---

In general case, I'd definitely prefer to have a full-functional slot content (that allows actual content, not render-text function). But as no app actually needed any true custom content, we can use text. It would make component more performant and simple.

Alternatively, we may add a new slot, but explicitly mark it as a custom content slot, not the default text content.